### PR TITLE
#9306 Support for CQL_FILTER in LayerDownload plugin

### LINKED ISSUE #9306: CQL_FILTER not supported for layer download
## Description

Layers can include in params the `cql_filter`. Anyway the download tool do not use it. 

## How to reproduce
-  Import a map like this [map_cql_filter.zip](https://github.com/geosolutions-it/MapStore2/files/12174423/map_cql_filter.zip)
- Try to download, via WPS, the data, toggling  "Download filtered dataset" on
- When downloaded, open the data with a GIS

*Expected Result*
The exported data is filtered as it is on the map (only california)

*Current Result*
The exported data is not filtered.

- [x] Not browser related

<details><summary> <b>Browser info</b> </summary>

(use this site: <a href="https://www.whatsmybrowser.org/">https://www.whatsmybrowser.org/</a> for non expert users)

| Browser Affected | Version |
|---|---|
|Internet Explorer| |
|Edge| |
|Chrome| |
|Firefox| |
|Safari| |
</details>

### PR DESCRIPTION
**Pull request by offtherailz**
## Description
This PR provides support to cql_filter in `LayerDownload` plugin. 

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

#9306 

**What is the new behavior?**
Fix #9306 

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/epics/layerdownload.js [hunk 1]
```diff
@@ -97,7 +97,10 @@ const DOWNLOAD_FORMATS_LOOKUP = {
 };
 
 const { getFeature: getFilterFeature, query, sortBy, propertyName } = requestBuilder({ wfsVersion: "1.1.0" });
-
+const getCQLFilterFromLayer = (layer = {}) => {
```
Comment (MV88) [inline web/client/epics/layerdownload.js:100]: - no jsdoc
- can this be moved into a utils file?

**Review by MV88 — CHANGES_REQUESTED** [review]
please check my comments inline

File: web/client/epics/layerdownload.js (hunk 1 above)
Comment (offtherailz) [reply web/client/epics/layerdownload.js:100]: I can document but I think It is confusing documenting private and self explaining functions that are used inline in a single file.
For the moment I should not export this. It is used only here.

Comment (offtherailz) [inline web/client/epics/layerdownload.js:100]:
```suggestion
/**
 * Extracts the `cql_filter` parameter from the layer object.
 * @private 
const getCQLFilterFromLayer = (layer = {}) => {
```

**Comment by MV88** [comment]
@ElenaGallo please test it in DEV

**Pull request merged**